File: Onoicrm.Front/src/modules/tooth/entities/BookingTooth.ts
import {BookingToothFile} from "~/modules/booking-tooth/entities/BookingToothFile";
import {Entity} from "~/shared/entities/Entity";

export class BookingTooth extends Entity{
  public caption: string = "";
  public description: string = "";
  public bookingId!: number
  public toothStateId!: number
  public toothId!: number
  public files:BookingToothFile[] = []
  public createDate!:string
  constructor(init?: Partial<BookingTooth>) {
    super();
    Object.assign(this, init)
  }
}

File: Onoicrm.Domain/Entities/BookingTooth.cs
using System.ComponentModel.DataAnnotations;

namespace Onoicrm.Domain.Entities;

public class BookingTooth : Entity
{
    [Required]
    public long ToothId { get; set; }
    public Tooth? Tooth { get; set; }
    [Required]
    public long BookingId { get; set; }
    public Booking? Booking { get; set; }
    public long? ToothStateId { get; set; }
    public ToothState? ToothState { get; set; }
    public string Caption { get; set; } = "";
    public string Description { get; set; } = "";
    public ICollection<BookingToothChannel> Channels { get; set; } = new List<BookingToothChannel>();
    public ICollection<BookingToothFile> Files { get; set; } = new List<BookingToothFile>();
    public ICollection<ImplementedService> ImplementedServices { get; set; } = new List<ImplementedService>();
    protected override int GetClassId() => ClassNames.ToothBooking;
}